### DISCUSSION TITLE
Breaking changes between v0.1.14 and v0.1.15 -- Removal of Console as a param

### DISCUSSION THREAD

**Discussion by M-Porter** [Textualize/textual#438] [General]
Why was `console` removed as a param in [this commit](https://github.com/Textualize/textual/commit/ffcbe4ba1e04f0bace7d7e8b87aa2e0390555b00)? I relied on being able to pass in console in order to to be able to run a textual cli app in a subshell. [This is how I use the param to be able to better control my output.](https://github.com/M-Porter/jumparound/blob/b4a73c4b011306f8a7462a5244daddcb7d9f3ff6/jumparound/cli.py#L28-L39).

If this was completely intentional with no intent to come back then I just want to know why because now I'll have to either hack around the library or completely rewrite my app in another tui lib (which would suck bc textual is awesome).

**Comment by jeffwright13** [comment]
I've had similar issues with older code not working on newer releases. The plain answer is that Textual is still under heavy development, so we're not going to be able to rely 100% on the stability of any of the code until they release the official version (which I gather will be within the next few months). Still, as you said, Textual is certainly the most compelling TUI out there that I've seen.

If you want to play with other TUIs, I can recommend Asciimatics, as well as PyTermTk (although that one is still in need of documentation). I've also had a bit of success with pytermgui and am intrigued by the Austin TUI. I have no interest in curses. :-)

**Reply by M-Porter** [reply]
I understand that they may have breaking changes and their APIs are unstable, but the removal of the old constructor for the new is a massive backwards step in being able to customize the library, especially if you compare the diff. The change has a useless commit message and is completely unrelated to the PR so I'm just trying to get clarification on the reasoning.

**Reply by willmcgugan** [reply]
The Console is an internal detail, and there is no need to expose it in the public interface.

This feels like a classic XY problem. Your are asking for feature Y to solve problem X. But what is X?

What problem was the Console was solving for you?

**Reply by M-Porter** [reply]
https://github.com/M-Porter/jumparound/blob/b4a73c4b011306f8a7462a5244daddcb7d9f3ff6/jumparound/cli.py#L28-L39

I wanted use stderr so that I could use textual in an app as a subshell. Without that param set to true on the console instance I could not do so.
